pkg/sdn/plugin/common.go

```diff
@@ -39,20 +39,36 @@ func clusterNetworkToString(n *osapi.ClusterNetwork) string {
 	return fmt.Sprintf("%s (network: %q, hostSubnetBits: %d, serviceNetwork: %q, pluginName: %q)", n.Name, n.Network, n.HostSubnetLength, n.ServiceNetwork, n.PluginName)
 }
 
+//determine if the first argument contains any of the second
+func cidrListContains(cidrList []*net.IPNet, ipaddr net.IP) (*net.IPNet, bool) {
+	for _, cidr := range cidrList {
+		if cidr.Contains(ipaddr) {
+			return cidr, true
+		}
+	}
+	return nil, false
+}
+
 type NetworkInfo struct {
-	ClusterNetwork *net.IPNet
+	ClusterNetwork []*net.IPNet
 	ServiceNetwork *net.IPNet
 }
 
-func parseNetworkInfo(clusterNetwork string, serviceNetwork string) (*NetworkInfo, error) {
-	cn, err := netutils.ParseCIDRMask(clusterNetwork)
-	if err != nil {
-		_, cn, err := net.ParseCIDR(clusterNetwork)
+func parseNetworkInfo(clusterNetwork []osapi.ClusterNetworkEntry, serviceNetwork string) (*NetworkInfo, error) {
+	var cn []*net.IPNet
+
+	for _, entry := range clusterNetwork {
```

Comment:
This is to verify there is no overlap between the given cidrs, otherwise pods may end up with same IPs.